Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkShaderCompatabilityUtility.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+namespace VRCAudioLink.Editor
+{
+    public class AudioLinkShaderCompatabilityUtility
+    {
+        private const string OldPath = "Assets/AudioLink/Shaders/AudioLink.cginc";
+        private const string NewPath = "Packages/com.llealloo.audiolink/Runtime/Shaders/AudioLink.cginc";
+        private const string MenuItemPath = "AudioLink/Upgrade AudioLink shaders";
+
+        private const string DialogText =
+            "Do you want to check all shaders in this project for AudioLink 0.3.x compatibility and update them if necessary? This is useful for upgrading projects using AudioLink 0.2.x or below."
+            + "\n" + "\n" +
+            "If you choose 'Go Ahead', shader files in this project which include 'AudioLink.cginc' will be edited to use the new path introduced by AudioLink 0.3.x. Shaders which already use the new path will be unaffected. You should make a backup before proceeding."
+            + "\n" + "\n" +
+            "(You can always run this utility manually via the " + MenuItemPath + " menu command)";
+
+        private const string DialogTitle = "AudioLink Shader Compatibility Utility";
+        private const string DialogOkButton = "I Made a Backup, Go Ahead!";
+        private const string DialogCancelButton = "No Thanks";
+
+
+        [MenuItem(MenuItemPath)]
+        public static void UpgradeShaders()
+        {
+            if (EditorUtility.DisplayDialog(DialogTitle, DialogText, DialogOkButton, DialogCancelButton))
+            {
+                UpgradeShaderFiles();
+                UpgradeCgincFiles();
+            }
+        }
+
+        private static void UpgradeShaderFiles()
+        {
+            ShaderInfo[] shaders = ShaderUtil.GetAllShaderInfo();
+
+            foreach (ShaderInfo shaderinfo in shaders)
+            {
+                Shader shader = Shader.Find(shaderinfo.name);
+                string path = AssetDatabase.GetAssetPath(shader);
+                // we want to avoid built-in shaders so we check the Path
+                if (path.StartsWith("Assets") || path.StartsWith("Packages"))
+                {
+                    ReplaceInFile(path);
+                }
+            }
+        }
+
+        private static void UpgradeCgincFiles()
+        {
+            List<string> cgincs = FindCgincFiles(Application.dataPath);
+            cgincs.AddRange(FindCgincFiles(Path.GetDirectoryName(Application.dataPath) + "/Packages"));
```

Comment:
What does Path.GetDirectoryName do when fed a directory path?

Reply:
It returns the parent folder